pkg/pod/pod.go

```diff
@@ -577,6 +577,52 @@ func (builder *Builder) WithTolerationToMaster() *Builder {
 	return builder
 }
 
+// WithToleration adds a toleration configuration inside the pod.
+func (builder *Builder) WithToleration(toleration v1.Toleration) *Builder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Redefining pod's %s with toleration %v", builder.Definition.Name, toleration)
+
+	builder.isMutationAllowed("custom toleration")
+
+	if builder.errorMsg != "" {
+		return builder
+	}
+
+	builder.Definition.Spec.Tolerations = append(builder.Definition.Spec.Tolerations, toleration)
```

Comment:
Given that we `append()`, maybe the glog above should say that. Redefine seems to mean _replace_ in most functions. 

Also if we want to append, maybe we should log the existing tolerations?

Reply:
Updated comment